istioctl/cmd/istioctl/rbac.go

```diff
@@ -84,6 +87,35 @@ istioctl experimental rbac can -s source.namespace=foo POST rating /data -a dest
 	return cmd
 }
 
+func upgrade() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "upgrade <rbac-file>",
+		Short: "Upgrade Istio Authorization policy file to use the newest API",
+		Long: `
+This command automatically converts Istio Authorization v1 policy files to v2. It requires the operator 
+to have READ access to the Kubernetes cluster where the services specified in ServiceRole definitions exist. 
+`,
+		Args: cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			upgrader, err := newUpgrader(args[0])
+			if err != nil {
+				return err
+			}
+			convertedPolicies, err := upgrader.UpgradeCRDs()
+			if err != nil {
+				return err
+			}
+			writer := os.Stdout
```

Comment:
Don't use `os.Stdout`.  Use `cmd.OutOrStdout()`.  That allows us to use Cobra test mechanisms.

Reply:
Sounds good. Will change. We probably need to change other places such as https://github.com/istio/istio/blob/8d82dbe6377f2ab8ea24a12b9ef0c0a1f7cd2f13/istioctl/cmd/istioctl/convert_ingress.go#L72 as well.